connectors/couchbase/translator-couchbase/src/main/java/org/teiid/translator/couchbase/CouchbaseMetadataProcessor.java

```diff
@@ -0,0 +1,718 @@
+
+    private boolean isArrayJsonType(Object jsonValue) {
+        return jsonValue instanceof JsonArray;
+    }
+    
+    /**
+     * For handle typed table name, replace the root keyspace to typed table name.
+     * @param name - typed table name.
+     * @param path - path of document
+     * @return
+     */
+    private String repleaceTypedName(String name, String path) {
+        String tableName = path.substring(path.indexOf(UNDERSCORE));
+        return name + tableName;
+    }
+    
+    private String buildArrayTableIdxName(String nameInSource, int dimension) {
+        StringBuilder sb = new StringBuilder();
+        String dim1Name = nameInSource.substring(0, nameInSource.indexOf(SQUARE_BRACKETS));
+        String[] names = dim1Name.split(Pattern.quote(SOURCE_SEPARATOR));
+        boolean isFirst = true;
+        for(String name : names) {
+            if(isFirst) {
+                isFirst = false;
+                sb.append(this.trimWave(name));
+            } else {
+                sb.append(UNDERSCORE);
+                sb.append(this.trimWave(name));
+            }
+        }
+
+        for(int i = 1 ; i <= dimension ; i ++) {
+            if(i == 1) {
+                continue;
+            }
+            sb.append(UNDERSCORE);
+            sb.append(DIM_SUFFIX).append(i);
+        }
+        sb.append(IDX_SUFFIX);
+        return sb.toString();
+    }
+    
+    private String buildNamedTypePair(String columnIdentifierName, String typedValue) {
+        StringBuilder sb = new StringBuilder();
+        sb.append(columnIdentifierName).append(COLON).append(QUOTE).append(typedValue).append(QUOTE); 
+        return sb.toString();
+    }
+    
+    private String getTypeName(String keyspace) {
+        
+        if(this.typeNameList == null) {
+            LogManager.logWarning(LogConstants.CTX_CONNECTOR, CouchbasePlugin.Util.gs(CouchbasePlugin.Event.TEIID29009));
+            return null;
+        }
+        
+        if(this.typeNameMap == null) {
+            this.typeNameMap = new HashMap<>();
+            try {
+                Pattern typeNamePattern = Pattern.compile(CouchbaseProperties.TPYENAME_MATCHER_PATTERN);
+                Matcher typeGroupMatch = typeNamePattern.matcher(typeNameList);
+                while (typeGroupMatch.find()) {
+                    typeNameMap.put(typeGroupMatch.group(1), typeGroupMatch.group(2));
+                }
+            } catch (Exception e) {
+                LogManager.logError(LogConstants.CTX_CONNECTOR, e, CouchbasePlugin.Util.gs(CouchbasePlugin.Event.TEIID29012, typeNameList));
+            }
+        } 
+
+        return this.typeNameMap.get(keyspace);
+    }
+    
+    protected void addProcedures(MetadataFactory metadataFactory, CouchbaseConnection connection) {
+        
+        Procedure getTextDocuments = metadataFactory.addProcedure(GETTEXTDOCUMENTS);
+        getTextDocuments.setAnnotation(CouchbasePlugin.Util.getString("getTextDocuments.Annotation")); //$NON-NLS-1$
```

Comment:
then you need to get rid of these and support the directly from transforming SQL statements.